sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -1280,6 +1280,14 @@ def test_enet_l1_ratio():
     with pytest.raises(ValueError, match=msg):
         MultiTaskElasticNetCV(l1_ratio=0, random_state=42).fit(X, y[:, None])
 
+    # Test that l1_ratio=0 with alpha>0 produces user warning
+    warning_message = (
+        "Coordinate descent with l1_reg=0 may lead to unexpected"
+        " results and is discouraged."
+    )
+    with pytest.warns(UserWarning, match=warning_message):
+        ElasticNetCV(l1_ratio=[0], alphas=[1]).fit(X, y)
```

Comment:
Since the user sets `l1_ratio` and the error mentions `l1_reg`, do you think the warning is still a little unclear?

Reply:
Hmm... will it be okay if I change to `"Coordinate descent with no regularization may lead to unexpected results and is discouraged."`, similar to the warning in `_cd_fast.enet_coordinate_descent`?

https://github.com/scikit-learn/scikit-learn/blob/35af6dc808c8d317eb7017d1b16e271c9c8bba77/sklearn/linear_model/_cd_fast.pyx#L148-L150

If we want to point out the individual `l1_ratio=0` or `alpha=0` instead, I guess we should refactor all the `enet` gradient descent methods in `linear_model._cd_fast`. Now they are taking in `l1_reg` and `l2_reg` but we can modify them to take in `alpha`, `l1_ratio` and `n_samples` instead?

Reply:
>  I guess we should refactor all the enet gradient descent methods 

I think a refactor would be too much of an effort just to adjust warning/error messages, so I would avoid it for this PR.

> will it be okay if I change to "Coordinate descent with no regularization may lead to unexpected results and is discouraged."

I like this idea. I think we can specify `L1` to keep the spirit of the original error message:

> Coordinate descent with no L1 regularization may lead to unexpected